engine/src/main/java/org/kigalisim/engine/support/EolEmissionsRecalcStrategy.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Strategy for recalculating end-of-life emissions.
+ *
+ * <p>This strategy encapsulates the logic previously found in the
+ * recalcEolEmissions method of SingleThreadEngine.</p>
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.support;
+
+import org.kigalisim.engine.Engine;
+import org.kigalisim.engine.number.EngineNumber;
+import org.kigalisim.engine.number.UnitConverter;
+import org.kigalisim.engine.state.OverridingConverterStateGetter;
+import org.kigalisim.engine.state.Scope;
+
+/**
+ * Strategy for recalculating end-of-life emissions.
+ */
+public class EolEmissionsRecalcStrategy implements RecalcStrategy {
+
+  private Scope scope;
+
+  /**
+   * Create a new EolEmissionsRecalcStrategy.
+   *
+   * @param scope The scope to use for calculations, null to use engine's current scope
+   */
+  public EolEmissionsRecalcStrategy(Scope scope) {
+    this.scope = scope;
+  }
+
+  @Override
+  public void execute(Engine target, RecalcKit kit) {
+    // Move the logic from SingleThreadEngine.recalcEolEmissions
+    // Setup
+    OverridingConverterStateGetter stateGetter =
+        new OverridingConverterStateGetter(kit.getStateGetter());
+    UnitConverter unitConverter = new UnitConverter(stateGetter);
+    Scope scopeEffective = scope != null ? scope : target.getScope();
+    String application = scopeEffective.getApplication();
+    String substance = scopeEffective.getSubstance();
+
+    // Check allowed
+    if (application == null || substance == null) {
+      ExceptionsGenerator.raiseNoAppOrSubstance("recalculating EOL emissions change", "");
+    }
+
+    // Calculate change
+    EngineNumber currentPriorRaw = target.getStreamRaw(application, substance, "priorEquipment");
+    EngineNumber currentPrior = unitConverter.convert(currentPriorRaw, "units");
+
+    stateGetter.setPopulation(currentPrior);
+    EngineNumber amountRaw = kit.getStreamKeeper()
```

Comment:
I see `kit.getStreamKeeper()` is needed in two places. Can we save it to a local variable to simplify?